x-pack/plugins/fleet/server/services/epm/archive/storage.ts

```diff
@@ -147,89 +139,32 @@ export async function getAsset(opts: {
 
   return storedAsset;
 }
+
 export const getEsPackage = async (
   pkgName: string,
   pkgVersion: string,
-  packageAssets: SavedObjectsBulkGetObject[],
+  packageAssets: PackageAssetReference[],
   savedObjectsClient: SavedObjectsClientContract
 ) => {
-  const pkgKey = pkgToPkgKey({ name: pkgName, version: pkgVersion });
-  const soRes = await savedObjectsClient.bulkGet<PackageAsset>(packageAssets);
-  const paths = soRes.saved_objects.map((asset) => asset.attributes.asset_path);
+  const bulkRes = await savedObjectsClient.bulkGet<PackageAsset>(packageAssets);
+  const entries: ArchiveEntry[] = bulkRes.saved_objects.map((so) => {
+    const { asset_path: path, data_utf8: utf8, data_base64: base64 } = so.attributes;
+    const buffer = utf8 ? Buffer.from(utf8, 'utf8') : Buffer.from(base64, 'base64');
 
-  // create the packageInfo
-  // TODO: this is mostly copied from validtion.ts, but we should save packageInfo somewhere
-  // so we don't need to do this again as this was already done either in registry or through upload
+    if (path && buffer) setArchiveEntry(path, buffer);
 
-  const manifestPath = `${pkgName}-${pkgVersion}/manifest.yml`;
-  const soResManifest = await savedObjectsClient.find<PackageAsset>({
-    type: ASSETS_SAVED_OBJECT_TYPE,
-    perPage: 1,
-    page: 1,
-    filter: `${ASSETS_SAVED_OBJECT_TYPE}.attributes.package_name:${pkgName} AND ${ASSETS_SAVED_OBJECT_TYPE}.attributes.package_version:${pkgVersion} AND ${ASSETS_SAVED_OBJECT_TYPE}.attributes.asset_path:${manifestPath}`,
+    return {
+      path,
+      buffer,
+    };
   });
-  const packageInfo = yaml.load(soResManifest.saved_objects[0].attributes.data_utf8);
+  preloadManifests(entries);
```

Comment:
We need to give all manifest files (root & data streams) to the validation functions.

This is all files in the package, but the validator filters the correct ones